caddyhttp/proxy/reverseproxy.go

```diff
@@ -200,7 +200,7 @@ func (rp *ReverseProxy) ServeHTTP(rw http.ResponseWriter, outreq *http.Request,
 		respUpdateFn(res)
 	}
 	if res.StatusCode == http.StatusSwitchingProtocols && strings.ToLower(res.Header.Get("Upgrade")) == "websocket" {
-		res.Body.Close()
+		defer res.Body.Close()
```

Comment:
The changes in https://github.com/mholt/caddy/commit/4babe4b201eef3e9794851b74f734a03338f7a20#diff-a71769529a289e215ba922059611256b seem to have removed the `defer` close call in favor of an explicit close call for the purpose of [populating trailers](https://github.com/mholt/caddy/commit/4babe4b201eef3e9794851b74f734a03338f7a20#diff-a71769529a289e215ba922059611256bR288). Do you want to add in the defer call higher still?